util/a2e/atoe_utils.c

```diff
@@ -440,7 +443,7 @@ atoe_vsnprintf(char *str, size_t count, const char *fmt, va_list args)
 			pattern = fmt - 1;                                 /*ibm@8665*/
 			left_justify = TRUE;
 			min_width = 0;
-			precision = this.end - this.buffer;
+			precision = INT_MAX;
```

Comment:
Why is initialization of precision is set to INT_MAX?

Reply:
In the implementation, this was used to limit the number of character we process. But to match the std vsnprintf, we need to process all chars therefore I set it to MAX. I decided to not remove the whole logic since it was used in cases that we want to limit the precision intentionally.

Reply:
Even in that case, do you expect `INT_MAX` number of characters ? This is definitely arbitrarily high number, and looking at the current logic, I do not see a way around that. That being said, I am OK if we have to refactor this heavily and also remove the precision (If not needed).

Reply:
Yes, so do we need that limit in the first place ?

Reply:
I rename that variable to `max_width` since it is not actually "precision" but the max chars to pick from the argument. I could use some conditions to set this only if it was specified but it makes the code a bit more complicated since more flags and condition codes would be needed. 
I think with the new name, the code make more sense. Basically it tells that the min width is 0 and max width is INT_MAX by default and we change the defaults if the format string requires it. 
I am not worried about the default value being too hi since without any boundary being specified, the max is infinite!

Reply:
`width` and `precision` are the accepted terminology for `printf` format specifier: `%[flags][width][.precision][length]specifier`.  We just overloaded the use of `precision` for handling `%s`'s limit.

Reply:
I propose we revert back to width and precision here.. and simply remove the use of `precision` for the `%s`/`fstring()` scenario.  No real reason why the variable needs to be overloaded for two separate use-cases.

Reply:
Agreed, there is no need to use `precision` the fstring case.

Reply:
I don't actually see the`precision` value being consumed for non `%s` cases.  For example, if we had `%4.2f`, we end up calling `snprintf` with the full pattern.

Reply:
As you mentioned, the `fstring` is the only consumer of `precision` but we can not remove it from there. for instance if we call `vsnprintf(buff, size, "%.2s", "ABCD")` we expect to get `"AB"` in buffer which is handled by `precision`. if I remove it it would generate `"ABCD"`!

Reply:
Okk, but having precision set to `INT_MAX` would also be incorrect, right? It is not incorrect to have buffer higher than `INT_MAX` passed into the function to fill in. In that case `INT_MAX` would limit filling the buffer till that point only. 

How about using `-1` precision suggesting uninitialized precision? Skimming through the uses, 
if it is with `%s` then and precision is negative, you read whole buffer, in case of `%.<n>s` the implementation sets the precision to 0 and recalculate it. 

In case of flongnumber and fnumber you have [place](https://github.com/ehsankianifar/omr/blob/386a7080ffa18909b7327e96eefa60c8db0be3e8/util/a2e/atoe_utils.c#L385) to control number of characters you read.

Reply:
sure, I'll use -1 as "not specified" indicator. the fnumber and flongnumber don't need precision. I've already removed that code from those methods (they just need width!)